assign-doi-datacite/src/main/java/no/unit/nva/datacite/mdsclient/DataCiteMdsConnection.java

```diff
@@ -0,0 +1,218 @@
+package no.unit.nva.datacite.mdsclient;
+
+import static nva.commons.utils.JsonUtils.objectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.ContentType;
+
+/**
+ * DataCiteMdsConnect instance for handling the HTTP communication with Datacite MDS API.
+ *
+ * <p>The HttpClient provided should have a {@link java.net.Authenticator} associated to do pre-emptive
+ * authentication towards the API server.
+ *
+ * <p>Use the {@link DataciteMdsConnectionFactory#getAuthenticatedConnection(String)} to construct new instances.
+ */
+public class DataCiteMdsConnection {
```

Comment:
This class has a lot of copy paste code. refactor to methods.

Reply:
Refactored with extracting a few methods.